### PR TITLE
Bcm bnxt drivers v1

### PR DESCRIPTION
**Pull request by sandeepbrcm**

### CODE REVIEW

File: core/drivers/bnxt/bnxt.c [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	    ((dst + num_entries * sizeof(uint32_t)) & BNXT_INDIRECT_BASE_MASK))
+		num_entries = (((dst + BNXT_INDIRECT_WINDOW_SIZE) &
+				BNXT_INDIRECT_BASE_MASK) -
+			       dst) /
+			      sizeof(uint32_t);
+
+	bnxt_prepare_access_window(dst);
+	target = bnxt_indirect_tgt_addr(dst);
+	for (i = 0; i < num_entries; i++) {
+		io_write32(target, *(uint32_t *)src);
```
Comment (sandeepbrcm) [inline core/drivers/bnxt/bnxt.c:74]: is this type cast needed.  *src  gives some warning ?

File: core/include/drivers/bcm-include/crc32.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	0xa9bcae53, 0xdebb9ec5, 0x47b2cf7f, 0x30b5ffe9,
+	0xbdbdf21c, 0xcabac28a, 0x53b39330, 0x24b4a3a6,
+	0xbad03605, 0xcdd70693, 0x54de5729, 0x23d967bf,
+	0xb3667a2e, 0xc4614ab8, 0x5d681b02, 0x2a6f2b94,
+	0xb40bbe37, 0xc30c8ea1, 0x5a05df1b, 0x2d02ef8d
+};
+
+#define ucrc32(ch, crc) (crc32tbl[(crc ^ (ch)) & 0xff] ^ (crc >> 8))
+
+static inline uint32_t crc32i(uint32_t crc, const char *buf, size_t len)
```
Comment (sandeepbrcm) [inline core/include/drivers/bcm-include/crc32.h]: pls google once if we use  CRC32  inbuilt instruction of A72  will it just work.  I mean read about CRC32 instruction of arm  cortex a72.   If its not straightforward lets stick to this routine and push to upstream for feedback.

**Review by sandeepbrcm — COMMENTED** [review]
Add proper commit message header eg: ..  driver: bnxt: add Broadcom bnxt driver

**Review by sandeepbrcm — COMMENTED** [review]
since this is the first pta to be pushed  bcmpta folder is created. lets push anyways and see if we are suggested to maintain it somewhere else

File: core/drivers/bnxt/bnxt.c (hunk 1 above)
Comment (shitalt) [reply core/drivers/bnxt/bnxt.c:74]: Yes, we get a error "core/drivers/bnxt/bnxt.c:74:22: error: invalid type argument of unary ‘*’ (have ‘uintptr_t {aka long unsigned int}’)"

**Pull request closed without merging**